kernel/smp.c

```diff
@@ -10,45 +10,41 @@
 #include <kswap.h>
 #include <kernel_internal.h>
 
-static struct k_spinlock global_spinlock;
-
-static volatile int recursive_count;
-
-/* FIXME: this value of key works on all known architectures as an
- * "invalid state" that will never be legitimately returned from
- * _arch_irq_lock().  But we should force the architecture code to
- * define something for us.
- */
-#define KEY_RECURSIVE 0xffffffff
+atomic_t global_lock;
```

Comment:
`global_lock` is a very generic name that shouldn't be visible outside this translation unit.  Can this be `static`?